txlib/api/tests/test_base_model.py

```diff
@@ -25,3 +34,80 @@ def test_join_subpaths(self):
         path = b._join_subpaths('/api/', '2/', 'projects/')
         assert path.count('/') == 4
         assert path == correct_path
+
+    def test_exception_if_wrong_url_field(self):
+        with pytest.raises(AttributeError):
+            BaseModel(invalid='anything')
+
+        with pytest.raises(AttributeError):
+            BaseModel(slug='slug', invalid='anything')
+
+    def test_exception_if_retrieving_invalid_attribute(self):
+        b = DummyModel(slug='slug')
+        with pytest.raises(AttributeError):
+            _ = b.something_invalid
+
+        # Existing attribute should work, however
+        assert b.slug == 'slug'
+
+    def test_get_populates_object(self):
+        setup_registry(get='{"id": 100, "param": "value"}')
+
+        obj = DummyModel.get(something='anything')
+        assert obj.id == 100
+        assert obj.param == 'value'
+        assert obj._populated_fields == {"id": 100, "param": "value"}
+
+    def test_setting_invalid_field_raises_error(self):
+        obj = DummyModel(slug='slug')
+        obj.name = 'name'
+        obj.description = 'description'
+        with pytest.raises(AttributeError):
+            obj.invalid_field = ''
+
+    def test_saving_existing_object_updates_it(self):
+        setup_registry(
+            get='{"id": 100, "param": "value"}',
+            put='{"slug": "slug", "name": "name"}',
+        )
+
+        obj = DummyModel.get(something='anything')
+        obj.slug = 'slug'
+        obj.name = 'name'
```

Comment:
Would the test also pass if we didn't set those two values?

Reply:
No, it fails with `AssertionError: assert None == 'slug'` and `AttributeError: DummyModel has no readable attribute name` respectively,